packages/cli/src/hooks/analytics/session.spec.ts

```diff
@@ -0,0 +1,117 @@
+jest.mock('../../lib/config/config');
+
+import type {IConfig} from '@oclif/config';
+import {mocked} from 'ts-jest/utils';
+import {Config} from '../../lib/config/config';
+import check from './session';
+
+describe('session', () => {
+  const mockedConfig = mocked(Config, true);
+  const mockedConfigGet = jest.fn();
+  const mockedConfigSet = jest.fn();
+  const theDate = new Date('1997-08-29T12:00:00Z');
+  const dummySessionId = 1234567;
+
+  const doMockConfig = () => {
+    mockedConfig.prototype.get = mockedConfigGet;
+    mockedConfig.prototype.set = mockedConfigSet;
+  };
+
+  const freezeTime = () => {
+    jest.useFakeTimers().setSystemTime(theDate.getTime());
+  };
+
+  const getTimeBefore = (minutes: number) => {
+    return theDate.getTime() - minutes * 60e3;
+  };
+
+  beforeAll(() => {
+    freezeTime();
+    global.config = {configDir: 'the_config_dir'} as IConfig;
```

Comment:
Mh, you should instead mock `Config` itself IMO.
This could lead to side-effects in other tests suite (current or future)

Reply:
Newbedev says
> As every test suite run its own environment, you can mock globals by just overwriting them. All global variables can be accessed by the global namespace [...] The overwrite has only effects in your current test and will not effect others.

If it's coming from Newbedev, it's necessarily true :P

Reply:
regardless, the `session` class, from a unit test standpoint should not care about the intricacy of the `Config` class.
This is why it's mocked in all the other unit test suites ;)